### DISCUSSION TITLE
How to display the table of content of each post ?

### DISCUSSION THREAD

**Discussion by louxyy** [CaiJimmy/hugo-theme-stack#1043] [Q&A]
Hello everyone 

I would like to display the table of content of each post. 
If i am not wrong, the table of contents is displayed via widget. 

According to the documentation on this [page](https://stack.jimmycai.com/config/widgets#toc) and taking this [line](https://github.com/CaiJimmy/hugo-theme-stack-starter/blob/df1f51edd7a8abc5913fe20dceeec78ee7e6d975/config/_default/params.toml#L55) as an example, I completed my "/config.toml"

```
baseURL = 'http://localhost'
languageCode = 'en-us'
title = "Tutoriel"
theme = "hugo-theme-stack"

[[widgets.homepage]]
type = "categories"

[[widgets.page]]
type = "toc"


```

But I don't still see a table of content to the right of my posts. Did I forget another parameter ?

**Comment by ElastiCourse** [comment]
Hi,
`config.toml` is a template. The actual configuration is in `hugo.yaml` or `hugo.toml` if you converted it.
These are the two sections involved in my `hugo.yaml`
```
    article:
        math: false
        toc: true
```
Make sure toc is set to true. In the same file:
```
    tableOfContents:
        endLevel: 4
        ordered: true
        startLevel: 2
```
This means table of contents will show starting from H2 all the way to H4 by order. You can modify that behavior if you like.
Finally, to make the sections appear in a page `content/post/my-blog-post/index.md` add the following:
```
---
Front matter here
---
## Section one
text
## Section two
text
### Section two sub a
text
### Section two sub b
text
## Section three
text
```
